### PR TITLE
Updates to Travis to resolve job timeouts.

### PR DESCRIPTION
**Pull request by aaroncblack**
Lower MPI ranks used in parallel examples to two, to match number of processor cores available in Travis VM.
Update code coverage c++ flags to add '-g' (mandatory)
Update code coverage travis job to be optimized code with debug symbols.
Only build Mac OpenMPI if parallel MFEM being used.
Lower -j4 to -j3, as two processor cores available.

### CODE REVIEW

**Comment by aaroncblack** [comment]
Running more MPI ranks than processor cores in the VM is noticeably slower.  I'm still suspicious that ex13p seems unusually vulnerable to hanging or taking an extremely long time when in that configuration.

The OSX jobs are not tested yet ( I made a small change to skip building the Openmpi when not needed ).  Travis queue is currently taking a long time, so will verify they passed Monday.

**Comment by aaroncblack** [comment]
I would be interested in seeing a run of ex13p on a LC node with 2x the number of MPI ranks as processor cores on the node.  See if that problem has any issues.

**Comment by tzanio** [comment]
It is really a bummer that we can only use 2 cores. Open MPI usually has no issues with overloading the cores, so I wonder if this is a due to the containerization. Should we consider going back to the pre-container version?

File: .travis.yml [hunk 1]
```diff
@@ -137,16 +126,15 @@ script:
      CPPFLAGS="" &&
      SKIP_TEST_DIRS="" &&
      if [ "$CODECOV" == "YES" ]; then
-        CPPFLAGS="--coverage";
+        CPPFLAGS="--coverage -g";
         SKIP_TEST_DIRS="miniapps/performance";
```
Comment (tzanio) [inline .travis.yml]: Is this still necessary? Maybe we can pick up a few more percentages if we don't skip the HPC examples?

**Comment by v-dobrev** [comment]
We can also try MPICH on ubuntu with 4 tasks - that may work better than OpenMPI.

File: .travis.yml (hunk 1 above)
Comment (v-dobrev) [reply .travis.yml]: May be worth a try to see how that looks.

**Comment by aaroncblack** [comment]
I switched to an optimized build with debug symbols, for the code cov so I'll bet I can re-enable the HPC mini app.

On September 16, 2017 at 3:40:22 PM PDT, Tzanio Kolev <notifications@github.com> wrote:

@tzanio commented on this pull request.

________________________________

In .travis.yml<https://github.com/mfem/mfem/pull/277#discussion_r139296883>:

> @@ -137,16 +126,15 @@ script:
      CPPFLAGS="" &&
      SKIP_TEST_DIRS="" &&
      if [ "$CODECOV" == "YES" ]; then
-        CPPFLAGS="--coverage";
+        CPPFLAGS="--coverage -g";
         SKIP_TEST_DIRS="miniapps/performance";


Is this still necessary? Maybe we can pick up a few more percentages if we don't skip the HPC examples?

**Comment by jedbrown** [comment]
You can oversubscribe by using MPICH with ch3:sock.  Nemesis is a
disaster when oversubscribed.

http://trac.mpich.org/projects/mpich/ticket/79

**Pull request merged**